src/app/features/task-repeat-cfg/store/task-repeat-cfg.effects.ts

```diff
@@ -58,6 +66,131 @@
+
+            const taskChanges = aAction.task.changes;
+
+            // TODO: is there a better way to do this? Is there anything missing?
+            const repeatCfgChanges: Partial<TaskRepeatCfgCopy> = {
+              projectId: taskChanges.projectId ?? repeatCfgForTask.projectId,
+              title: taskChanges.title ?? repeatCfgForTask.title,
+              tagIds: taskChanges.tagIds ?? repeatCfgForTask.tagIds,
+              notes: taskChanges.notes ?? repeatCfgForTask.notes,
+            };
+
+            // TODO: Do we need to do this for all instances??
+            this._taskRepeatCfgService.updateTaskRepeatCfg(
+              task.repeatCfgId,
+              repeatCfgChanges,
+            );
+          }
+        }),
+      ),
+    { dispatch: false }, // Question: What exactly does this do?
+  );
+
+  /**
+   * When a main task is made repeatable, this function checks if there are subtasks.
+   * If that is the case, a repeat-cfg gets added for each subtask, too.
+   */
+  addTaskRepeatCfgForSubTasksOf: any = createEffect(
+    () =>
+      this._actions$.pipe(
+        ofType(addTaskRepeatCfgToTask),
+        tap(async (aAction) => {
+          // TODO: is there an easier way to get to the parent task?
+          const allTasks = await this._taskService.allTasks$.pipe(first()).toPromise();
+          const parentTask = allTasks.find((aTask) => aTask.id === aAction.taskId);
+          const parentTaskRepeatCfg = aAction.taskRepeatCfg;
+
+          if (parentTask !== undefined && parentTask.subTaskIds.length > 0) {
+            for (const aSubTaskId of parentTask.subTaskIds) {
+              const task = allTasks.find((aTask) => aTask.id === aSubTaskId)!;
+
+              const repeatCfg = {
+                ...parentTaskRepeatCfg,
+                // TODO: anything missing in this list that should not be overwritten by the parent?
+                title: task.title,
+                notes: task.notes,
+                defaultEstimate: task.timeEstimate, // is this correct?
+                parentId: parentTask.repeatCfgId,
+              };
+
+              this._taskRepeatCfgService.addTaskRepeatCfgToTask(
+                task.id,
+                task.projectId,
+                repeatCfg,
+              );
+            }
+          }
+        }),
+      ),
+    { dispatch: false }, // Question: What exactly does this do?
+  );
+
+  /**
+   * When adding a sub task, this function checks if the parent is a repeatable task and therefore the sub-task also has to be.
+   * If that is the case, a repeat-cfg gets added for each subtask, too.
+   */
+  addTaskRepeatCfgForSubTask: any = createEffect(
+    () =>
+      this._actions$.pipe(
+        ofType(addSubTask),
+        tap(async (aAction) => {
+          const task = aAction.task;
+
+          // we only want to continue if the task doesn't already have a repeatCfgId
+          if (task.repeatCfgId === null) {
+            console.log('A TASKKK', task);
```

Comment:
Consider removing the console.log statements as they are generally not recommended for production code. They can expose sensitive information and clutter the console output. [important]